build_tools/azure/update_environments_and_lock_files.py

```diff
@@ -0,0 +1,412 @@
+"""Script to update CI environment files and associated lock files.
+
+To run it you need to be in the root folder of the scikit-learn repo:
+python build_tools/azure/update_environments_and_lock_files.py
+
+Two scenarios where this script can be useful:
+- make sure that the latest versions of all the dependencies are used in the CI.
+  We can run this script regularly and open a PR with the changes to the lock
+  files. This workflow will eventually be automated with a bot in the future.
+- bump minimum dependencies in sklearn/_min_dependencies.py. Running this
+  script will update both the CI environment files and associated lock files.
+  You can then open a PR with the changes.
```

Comment:
Can we add one more scenario resolving issues with `default_package_constraints`?

Reply:
Done